### ISSUE TITLE
Is the deploy.sh still valid?

### ISSUE THREAD

**Issue by FeynmanZhou** [OpenFunction/OpenFunction#193]
As `ofn` CLI has been officially recommended by the OpenFunction community and old install/uninstall scripts deprecation was mentioned in v0.5.0 release notes, is the [deploy.sh](https://github.com/OpenFunction/OpenFunction/blob/main/hack/deploy.sh)  and [delete.sh](https://github.com/OpenFunction/OpenFunction/blob/main/hack/delete.sh) still valid for the old version of OpenFunction?

**Comment by benjaminhuo** [comment]
It's still valid but not recommended anymore.
It's not removed because CI is still using these.
@tpiperatgod Do we have a plan to replace deploy.sh/deleted.sh with ofn in CI? You mentioned we need changes to ofn?

**Comment by tpiperatgod** [comment]
To do that, we need an external environment to execute the GitHub action.

**Comment by FeynmanZhou** [comment]
> 

As I know, SIG-DevOps KubeSphere is using Tekton in [ks-devops ](https://github.com/kubesphere/ks-devops)CI with GitHub action. @tpiperatgod You can ask @Linuxsuren @johnniang for details.

**Comment by arugal** [comment]
It looks like just need to add a `action.yaml` file to the https://github.com/OpenFunction/cli and use `ofn` like this:
```yaml
- name: Install dependent components
  uses: OpenFunction/cli@main 
  with:
    all: true
```

**Comment by tpiperatgod** [comment]
> It looks like just need to add a [action.yaml](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions) file to the [cli](https://github.com/OpenFunction/cli) and use `ofn` like this:
> 
> ```yaml
> - name: Install dependent components
>   uses: OpenFunction/cli@main 
>   with:
>     all: true
> ```

yes, but it failed during installation and we think it may be a lack of sufficient resources..

**Comment by benjaminhuo** [comment]
> > It looks like just need to add a [action.yaml](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions) file to the [cli](https://github.com/OpenFunction/cli) and use `ofn` like this:
> > ```yaml
> > - name: Install dependent components
> >   uses: OpenFunction/cli@main 
> >   with:
> >     all: true
> > ```
> 
> yes, but it failed during installation and we think it may be a lack of sufficient resources..

Regarding the limited resource issue, @wanjunlei handle it this way in the script approach: https://github.com/OpenFunction/OpenFunction/blob/main/.github/workflows/main.yaml#L81
